include/fmt/ostream.h

```diff
@@ -10,6 +10,10 @@
 
 #include <fstream>
 #include <ostream>
+#if defined(_WIN32) && defined(__GLIBCXX__)
+#  include <ext/stdio_filebuf.h>
+#  include <ext/stdio_sync_filebuf.h>
+#endif
```

Comment:
Why? In case some future GCC removes these files? The checks for Windows and libstdc++ should be still there.

Reply:
If the files are remove the check will be broken in either case but the removal is easier to support with `FMT_HAS_INCLUDE`. In general testing against features is more robust than testing against platforms.

Reply:
So I suggestion something like

```
#if defined(_WIN32) && FMT_HAS_INCLUDE(<ext/stdio_filebuf.h>)
#  include <ext/stdio_filebuf.h>
#  include <ext/stdio_sync_filebuf.h>
#  define FMT_USE_STDIO_FILEBUF
#endif
```

and later:

```
#ifdef FMT_USE_STDIO_FILEBUF
...
#endif
```

This will avoid duplicating checks and make the whole thing more robust (including to removal of headers).

Reply:
That will incur additional penalty during compilation when one does not use libstdc++ on Windows, e.g. in MSVC. There is additional filesystem access. The header `ext/stdio_filebuf.h` is very much platform specific thing so it is better to check for platform. Also think of the scenario when someone under MSVC somehow puts `ext/stdio_filebuf.h`. The check for the platform must be there, IMO.

See
1. https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_macros.html
2. https://gcc.gnu.org/onlinedocs/libstdc++/manual/ext_io.html
3. https://gcc.gnu.org/onlinedocs/gcc-12.1.0/libstdc++/api/a01178.html

> Beginning with 3.1, the extra basic_filebuf constructor and the fd() function were removed from the standard filebuf. Instead, <ext/stdio_filebuf.h> contains a derived class template called __gnu_cxx::stdio_filebuf. This class can be constructed from a C FILE* or a file descriptor, and provides the fd() function.

Its there since GCC v3.1 and will remain there because it is stable extension (not experimental). The older supported GCC compiler of FMT is v4.8.

Reply:
Fair enough.